airflow/models/taskinstance.py

```diff
@@ -3892,9 +3892,12 @@ def schedule_downstream_tasks(self, session: Session = NEW_SESSION, max_tis_per_
 
         :meta: private
         """
-        return TaskInstance._schedule_downstream_tasks(
-            ti=self, session=session, max_tis_per_query=max_tis_per_query
-        )
+        try:
+            return TaskInstance._schedule_downstream_tasks(
+                ti=self, session=session, max_tis_per_query=max_tis_per_query
+            )
+        except Exception:
+            self.log.exception("Error scheduling downstream tasks. Skipping it")
```

Comment:
Does this call for action from user side?
The thing about logging such exceptions is that users want to solve the underlying issue but from the description it sounds like there is no real action item to avoid it? Should we leave hints in the message (for example consider disable mini scheduler)

Reply:
Yeah, this could confuse the users. We could pass and not log anything since the users can't do anything about it, and the scheduler will still schedule the given task.

Reply:
I **think** it's still useful to know that it's happening - the main thing it currently struggles with is the exit code that is now 0. And it was quite confusing because task succeeded, but the process failed.

I imagine this log will provide some useful information - for example if we just silently discard it in https://github.com/apache/airflow/issues/39717  - we would never have known that the problem is with serializing. Which **might** be something theat the user can do something about - we have no idea, because there might be varioue reasons for the exception.

I believe it is quite good to explicitly tell "something happen" - and while yes - we do not know exactly what the user can do to avoid it, it will at least give the user a chance to understand and fix it (or maybe submit an issue / discussion here and get help.

I am generally very, very cautious about totally ignoring and silencing such exceptions - because we have no idea what's going on. In this case we decided that we won't do a confusing thing - because mini-scheduler is purely optional we can just not fail the process, but having a log is still useful.

Reply:
> I believe it is quite good to explicitly tell "something happen" - and while yes - we do not know exactly what the user can do to avoid it, it will at least give the user a chance to understand and fix it (or maybe submit an issue / discussion here and get help.

Then lets say this explicitly in the message. I think we can just say that it can be caused by several reasons and link to the relevant discussion.

Reply:
Good idea.

Reply:
Done.

Reply:
@ephraimbuddy @eladkal -> see if you like it